src/app/lib/analyticsUtils/chartbeat.test.js

```diff
@@ -0,0 +1,158 @@
+import {
+  chartbeatUID,
+  useCanonical,
+  getSylphidCookie,
+  getDomain,
+  buildSections,
+  getType,
+} from './chartbeat';
+import onClient from '../utilities/onClient';
+
+let isOnClient = false;
+
+jest.mock('../utilities/onClient', () => jest.fn());
+onClient.mockImplementation(() => isOnClient);
+
+describe('Chartbeat utilities', () => {
+  it('should return the correct chartbeat UID', () => {
+    expect(chartbeatUID).toBe(50924);
+  });
+
+  it('useCanonical should be true', () => {
+    expect(useCanonical).toBe(true);
+  });
+
+  describe('Chartbeat ID Cookie', () => {
+    it('should return null when onClient is false', () => {
+      expect(getSylphidCookie()).toBe(null);
+    });
+
+    it('should return null when ID cookie does not exist', () => {
+      isOnClient = true;
+      expect(getSylphidCookie()).toBe(null);
+    });
+
+    it('should return the contents of the ID cookie when a value is present', () => {
+      const expectedCookieValue = 'foobar';
+      isOnClient = true;
+
+      Object.defineProperty(window.document, 'cookie', {
+        writable: true,
+        value: `ckns_sylphid=${expectedCookieValue}`,
+      });
+
+      expect(getSylphidCookie()).toBe(expectedCookieValue);
+    });
+  });
+
+  describe('Chartbeat Page Type', () => {
+    const types = [
+      {
+        type: 'article',
+        expectedType: 'New Article',
+      },
+      {
+        type: 'index',
+        expectedType: 'Index',
+      },
```

Comment:
I'd like another case to confirm intended behaviour when `type` is neither.

Reply:
Have implemented